beluga/include/beluga/sensor/data/occupancy_grid.hpp

```diff
@@ -0,0 +1,177 @@
+// Copyright 2023 Ekumen, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef BELUGA_SENSOR_DATA_OCCUPANCY_GRID_HPP
+#define BELUGA_SENSOR_DATA_OCCUPANCY_GRID_HPP
+
+#include <cstdint>
+#include <tuple>
+#include <vector>
+
+#include <beluga/sensor/data/linear_grid.hpp>
+
+#include <range/v3/view/enumerate.hpp>
+#include <range/v3/view/filter.hpp>
+#include <range/v3/view/transform.hpp>
+
+#include <Eigen/Core>
+
+namespace beluga {
+
+/**
+ * \page OccupancyGrid2dPage Beluga named requirements: OccupancyGrid2d
```

Comment:
There are some documentation warnings:
```suggestion
 * \page OccupancyGrid2Page Beluga named requirements: OccupancyGrid2d
```

JIC, you can build docs locally with: `src/beluga/docs/generate_docs.sh`. Here are some warnings:
```
/ws/src/beluga/beluga/include/beluga/algorithm/raycasting.hpp:56: warning: argument 'OccupancyGrid' of command @param is not found in the argument list of beluga::Ray2d< OccupancyGrid, Algorithm >::Ray2d(const OccupancyGrid &grid, Algorithm algorithm, const Sophus::SE2d &source_pose, double max_range)
/ws/src/beluga/beluga/include/beluga/algorithm/raycasting.hpp:62: warning: The following parameter of beluga::Ray2d::Ray2d(const OccupancyGrid &grid, Algorithm algorithm, const Sophus::SE2d &source_pose, double max_range) is not documented:
  parameter 'grid'
/ws/src/beluga/beluga/include/beluga/sensor/data/dense_grid.hpp:93: warning: Member index_at(const Eigen::Vector2i &pi) const (function) of class beluga::BaseDenseGrid2 is not documented.
/ws/src/beluga/beluga/include/beluga/sensor/data/occupancy_grid.hpp:68: warning: unable to resolve reference to 'OccupancyGrid2Page' for \ref command
```

Reply:
Indeed. I got bit by this twice already. It would be beneficial to have all CI checks bound to `colcon test` or as a pre-commit hook, to avoid the surprises.

Reply:
I agree.